client/network/sync/src/lib.rs

```diff
@@ -1362,6 +1363,10 @@ where
 		}
 		self.process_outbound_requests();
 
+		if let Some(warp_sync) = &mut self.warp_sync {
+			let _ = warp_sync.poll(cx);
```

Comment:
This must be called before `process_outbound_requests()`. If `poll` detects a target block it will have the request ready right away. Otherwise the request will only be issued on the next `poll` call, which may never come.

Reply:
Ah interesting, great catch @arkpar. Will resolve shortly